### PR TITLE
Ubuntu 22.04, first batch and misc

### PR DESCRIPTION
**Pull request by nobodysu**

### CODE REVIEW

File: apparmor.d/groups/bus/dbus-daemon [hunk 1]
```diff
@@ -11,6 +11,10 @@ include <tunables/global>
 profile dbus-daemon @{exec_path} flags=(attach_disconnected) {
   include <abstractions/base>
   include <abstractions/consoles>
+  include <abstractions/dbus>
```
Comment (nobodysu) [inline apparmor.d/groups/bus/dbus-daemon:14]: Extensive access to DBus, since it's DBus itself.

File: apparmor.d/profiles-a-f/agetty [hunk 2]
```diff
@@ -25,7 +26,9 @@ profile agetty @{exec_path} {
   owner @{run}/agetty.reload rw,
         @{run}/resolvconf/resolv.conf r,
 
-  /dev/tty[0-9]* rw,
+        /dev/tty[0-9]*   rw,
+  owner /dev/ttyGS[0-9]* rw,
```
Comment (nobodysu) [inline apparmor.d/profiles-a-f/agetty:30]: On some VPSes and hardware.

File: apparmor.d/profiles-g-l/logrotate [hunk 3]
```diff
@@ -36,7 +37,8 @@ profile logrotate @{exec_path} flags=(attach_disconnected, complain) {
   /{usr/,}bin/zstd                   rix,
   /{usr/,}{s,}bin/invoke-rc.d        rix,
   /{usr/,}lib/rsyslog/rsyslog-rotate rix,
-  /{usr/,}bin/fail2ban-client        rPx,
+  /{usr/,}bin/fail2ban-client                rPx,
+  /{usr/,}bin/systemd-tty-ask-password-agent rPx,
```
Comment (nobodysu) [inline apparmor.d/profiles-g-l/logrotate:41]: Rabbit hole, nowhere near finished.

File: apparmor.d/profiles-m-r/rsyslogd [hunk 4]
```diff
@@ -26,8 +26,14 @@ profile rsyslogd @{exec_path} {
   # for creating new log files and changing their owner/group
   capability chown,
 
+  # downgrade privileges on Ubuntu
+  capability setgid,
```
Comment (nobodysu) [inline apparmor.d/profiles-m-r/rsyslogd:30]: Won't boot otherwise.

File: apparmor.d/groups/bus/dbus-daemon [hunk 5]
```diff
@@ -63,12 +66,11 @@ profile dbus-daemon @{exec_path} flags=(attach_disconnected) {
 
   owner /tmp/dbus-[0-9a-zA-Z]* rw,
 
-  owner @{run}/user/@{uid}/bus w,
```
Comment (nobodysu) [inline apparmor.d/groups/bus/dbus-daemon:66]: Abstraction.

File: apparmor.d/groups/bus/dbus-daemon [hunk 6]
```diff
@@ -41,7 +45,6 @@ profile dbus-daemon @{exec_path} flags=(attach_disconnected) {
   /{usr/,}lib/dbus-1.0/dbus-daemon-launch-helper rPx,
 
   /etc/dbus-1/{,**} r,
-  /etc/machine-id r,
```
Comment (nobodysu) [inline apparmor.d/groups/bus/dbus-daemon:44]: Abstraction.

File: apparmor.d/groups/systemd/systemd-logind [hunk 7]
```diff
@@ -64,10 +65,12 @@ profile systemd-logind @{exec_path} flags=(attach_disconnected,complain) {
   @{run}/systemd/seats/ rw,
   @{run}/systemd/seats/.#seat* rw,
   @{run}/systemd/seats/seat[0-9]* rw,
-  @{run}/systemd/sessions/ rw,
-  @{run}/systemd/sessions/.#* rw,
-  @{run}/systemd/sessions/[0-9]*{,.ref} rw,
+  @{run}/systemd/sessions/      rw,
+  @{run}/systemd/sessions/*     r,
```
Comment (nobodysu) [inline apparmor.d/groups/systemd/systemd-logind:69]: We'll see how it goes.

File: apparmor.d/groups/systemd/systemd-logind [hunk 8]
```diff
@@ -105,5 +108,56 @@ profile systemd-logind @{exec_path} flags=(attach_disconnected,complain) {
   /dev/shm/{,**/} rw,
   /dev/tty[0-9]* rw,
 
+  # DBus
```
Comment (nobodysu) [inline apparmor.d/groups/systemd/systemd-logind:111]: I'm trying not to give logind access to whole DBus. Afraid it's just an introduction.

**Comment by nobodysu** [comment]
https://github.com/roddhjav/apparmor.d/blob/8deddc8a2c8862c83f2dd9171bdfcab3a80cdb4c/apparmor.d/groups/systemd/systemd-logind#L104

Raw access to block device? Is it really needed?

File: apparmor.d/profiles-a-f/agetty [hunk 9]
```diff
@@ -15,17 +15,24 @@ profile agetty @{exec_path} {
...
   capability sys_admin,
   capability sys_tty_config,
+  capability chown,
 
   @{exec_path} mr,
 
   /{usr/,}bin/login rPx,
 
-  /etc/issue r,
+  /{etc,run,lib,usr/lib}/issue        r,
```
Comment (nobodysu) [inline apparmor.d/profiles-a-f/agetty:24]: From man.

File: apparmor.d/profiles-g-l/login [hunk 10]
```diff
@@ -18,20 +18,46 @@ profile login @{exec_path} {
   capability fsetid,
   capability setgid,
   capability setuid,
+  capability sys_resource,
+  capability audit_write,
+  capability dac_read_search,
+#  capability net_admin,
+
+#  network netlink raw,
```
Comment (nobodysu) [inline apparmor.d/profiles-g-l/login:26]: At least local tty login and remote login.

File: apparmor.d/profiles-g-l/logrotate [hunk 11]
```diff
@@ -61,7 +67,8 @@ profile logrotate @{exec_path} flags=(attach_disconnected, complain) {
   /var/lib/logrotate.status rwk,
   /var/lib/logrotate.status.tmp rw,
 
-  /var/log/** rw,
+  /var/log{,.hdd}/   r,
+  /var/log{,.hdd}/** rw,
```
Comment (nobodysu) [inline apparmor.d/profiles-g-l/logrotate:71]: Zram setups.

File: apparmor.d/profiles-g-l/logrotate [hunk 12]
```diff
@@ -45,12 +50,15 @@ profile logrotate @{exec_path} flags=(attach_disconnected, complain) {
   include <abstractions/wutmp>
   ptrace (read),
   capability sys_ptrace,
+#  capability net_admin,
```
Comment (nobodysu) [inline apparmor.d/profiles-g-l/logrotate:53]: Called by `systemctl`.

**Comment by roddhjav** [comment]
Thank for the PR. Merged!

I am also working on Ubuntu 22.04. Especially on dbus rules. So some part may evolves later.

**Comment by nobodysu** [comment]
I'm looking forward to it.
And what's your opinion?

> https://github.com/roddhjav/apparmor.d/blob/8deddc8a2c8862c83f2dd9171bdfcab3a80cdb4c/apparmor.d/groups/systemd/systemd-logind#L104
> 
> Raw access to block device? Is it really needed?

**Comment by roddhjav** [comment]
I forgot this. No need to have it here, it is already included in the `disks-write` abstraction.

**Pull request merged**